### PR TITLE
node 11.0.0, node@10 10.12.0 (new formula)

### PR DESCRIPTION
**Pull request by dio-az**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/node@10.rb [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  fails_with :clang if MacOS.version <= :snow_leopard
+  fails_with :gcc_4_0
+  fails_with :gcc_4_2
+  ("4.3".."4.7").each do |n|
+    fails_with :gcc => n
+  end
+
+  # We track major/minor from upstream Node releases.
+  # We will accept *important* npm patch releases when necessary.
+  resource "npm" do
```
Comment (diogoazevedos) [inline Formula/node@10.rb]: Based on others versioned formulas, should I remove this?

File: Formula/node@10.rb [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    sha256 "a48e0b4471d72936afb598ebde0e07076598ac8647c2e9ebe891db5d6fbf2952"
+  end
+
+  def install
+    # Never install the bundled "npm", always prefer our
+    # installation from tarball for better packaging control.
+    args = %W[--prefix=#{prefix} --without-npm]
+    args << "--with-intl=system-icu" if build.with? "icu4c"
+    args << "--shared-openssl" << "--openssl-use-def-ca-store" if build.with? "openssl@1.1"
+    args << "--tag=head" if build.head?
```
Comment (diogoazevedos) [inline Formula/node@10.rb]: Should I remove this one also?

File: Formula/node@10.rb [hunk 3]
```diff
@@ -0,0 +1,138 @@
+class NodeAT10 < Formula
+  desc "Platform built on V8 to build network applications"
+  homepage "https://nodejs.org/"
+  url "https://nodejs.org/dist/v10.12.0/node-v10.12.0.tar.gz"
+  sha256 "c6552b95062f5e9f3a27da6fdb57914ab4b27a9aa2e783fb050791166554d059"
+  head "https://github.com/nodejs/node.git"
```
Comment (diogoazevedos) [inline Formula/node@10.rb]: Remove this spec and add the `keg_only :versioned_formula`.
```
Versioned formulae should use `keg_only :versioned_formula`
Formulae should not have a `HEAD` spec
```

**Comment by commitay** [comment]
`generate-json-schema` test failure on mojave is known.

`joplin` just seems to need a revision bump but I can do that in another PR so we don't need to rebuild this.

**Comment by commitay** [comment]
Thanks @diogoazevedos!

**Pull request closed without merging**